x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -276,49 +315,37 @@ private void updateEngine(Engine engine, ActionListener<IndexResponse> listener)
         }
     }
 
-    /**
-     * Deletes the provided {@param engineName} in the underlying index, or delegate a failure to the provided
-     * listener if the resource does not exist or failed to delete.
-     *
-     * @param engineName The name of the {@link Engine} to delete.
-     * @param listener The action listener to invoke on response/failure.
-     *
-     */
-    public void deleteEngine(String engineName, ActionListener<DeleteResponse> listener) {
-        try {
-            // TODO Delete alias when Engine is deleted
-            final DeleteRequest deleteRequest = new DeleteRequest(ENGINE_ALIAS_NAME).id(engineName)
-                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-            clientWithOrigin.delete(deleteRequest, listener);
-        } catch (Exception e) {
-            listener.onFailure(e);
-        }
-    }
-
     /**
      * List the {@link Engine} in ascending order of their names.
      *
      * @param queryString The query string to filter the results.
      * @param from From index to start the search from.
      * @param size The maximum number of {@link Engine} to return.
      * @param listener The action listener to invoke on response/failure.
+     *  It will receive a tuple containing the Engine results and an Integer with the total number of engines found
      */
-    public void listEngine(String queryString, int from, int size, ActionListener<SearchResponse> listener) {
+    public void listEngines(String queryString, int from, int size, ActionListener<Tuple<Engine[], Integer>> listener) {
         try {
             final SearchSourceBuilder source = new SearchSourceBuilder().from(from)
                 .size(size)
                 .query(new QueryStringQueryBuilder(queryString))
                 .docValueField(NAME_FIELD.getPreferredName())
                 .docValueField(INDICES_FIELD.getPreferredName())
+                .docValueField(ANALYTICS_COLLECTION_NAME_FIELD.getPreferredName())
                 .storedFields(Collections.singletonList("_none_"))
                 .sort(NAME_FIELD.getPreferredName(), SortOrder.ASC);
             final SearchRequest req = new SearchRequest(ENGINE_ALIAS_NAME).source(source);
-            clientWithOrigin.search(req, listener);
+            clientWithOrigin.search(req, listener.map(EngineIndexService::mapSearchResponse));
         } catch (Exception e) {
             listener.onFailure(e);
         }
     }
 
+    private static Tuple<Engine[], Integer> mapSearchResponse(SearchResponse response) {
```

Comment:
Agreed - I like the abstraction here (`listEngines()` returns a list of `Engine`s, regardless of where they came from).

Also, is it a convention to use `Tuple`s for compound return types, rather than introducing a new class?

Reply:
I agree with @demjened , a private class might feel a bit cleaner here.

Reply:
Sounds good! I will implement that as a static private class. Thanks! 👍